feat: I18n Scheme (#17768)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

Use the `extends` field instead of this if, for example, you need the ability for a repository config (e.g. `renovate.json`) to be able to use `ignorePresets` for any preset defined in global config.

## locale

Renovate supports internationalization with [GNU Gettext](https://www.gnu.org/software/gettext/), you can specific expected locale with this option, and also assign translationsFilePath is required.

```sh
--locale fr

> ```

Reviewer: should be a full sample or removed

## token

## translationsFilePath

Renovate supports internationalization with [GNU Gettext](https://www.gnu.org/software/gettext/), you can specific expected locale with this option, and also assign translationsFilePath is required.

This option is specific the PO(Portable Object) file's path.

> Renovate uses the default single domain "messages" for i18n to decline the complexity of usage of it.

Reviewer 2: ```suggestion
Renovate uses the default single domain "messages" for i18n to define the complexity of usage of it.
```

> How to contribute for i18n of Renovate ? Here is the workflow looks like:

Reviewer 2: ```suggestion
Here is the workflow for contributing an i18n translation for Renovate:
```

> Firt, you have to invok/mark string literals that you want to translate, and then invoke `gettext` function (or its siblings, e.g. `ngettext`, `pgettext` etc.) on them, for instance:

Reviewer 2: ```suggestion
First, you have to wrap string literals that you want to translate, and then invoke `gettext` function (or its siblings, e.g. `ngettext`, `pgettext` etc.) on them, for instance:
```

This is too much detail for the self-hosted configuration file

```typescript
{

>   description: gt._(

Reviewer 2: ```suggestion
  description: _(
```

## token

## translationsFilePath

This option is specific the PO(Portable Object) file's path.

Renovate uses the default single domain "messages" for i18n.

> You can translate your own edition by inovking `msginit` program, and then invoke `msgmerge` program to add more `msgid`.

Reviewer 2: I think that this needs to link to a fuller document about how to create own translations, otherwise one sentence is too little
